src/containers/Application/NavigationBox.tsx

```diff
@@ -0,0 +1,146 @@
+        <Grid.Column>
+          {!isLastPage ? (
+            <PageButton
+              title="Next"
+              type="right"
+              onClicked={() => nextPageButtonHandler(changePageProps)}
+            />
+          ) : (
+            <PageButton
+              title="Summary"
+              type="right"
+              onClicked={() => nextPageButtonHandler(changePageProps)}
+            />
+          )}
+        </Grid.Column>
+      </Grid.Row>
+    </Grid>
+  ) : null
+}
+
+interface checkPageProps {
+  sectionCode: string
+  currentPage: number
+  templateSections: TemplateSectionPayload[]
+}
+
+function checkFirstPage({ sectionCode, currentPage, templateSections }: checkPageProps): boolean {
+  const previousPage = currentPage - 1
+  const currentSection = templateSections.find(({ code }) => code === sectionCode)
+  if (!currentSection) {
+    console.log('Problem to find currentSection!')
+    return true
+  }
+  return previousPage > 0 ||
+    (previousPage === 0 && templateSections.find(({ index }) => index === currentSection.index - 1))
+    ? false
+    : true
+}
+
+function checkLastPage({ sectionCode, currentPage, templateSections }: checkPageProps): boolean {
+  const nextPage = currentPage + 1
+  const currentSection = templateSections.find(({ code }) => code === sectionCode)
+  if (!currentSection) {
+    console.log('Problem to find currentSection!')
+    return true
+  }
+  return nextPage <= currentSection.totalPages ||
+    (nextPage > currentSection.totalPages &&
+      templateSections.find(({ index }) => index === currentSection.index + 1))
+    ? false
+    : true
+}
+
+interface changePageProps {
+  currentPage: number
+  currentSection: TemplateSectionPayload
+  templateSections: TemplateSectionPayload[]
+  sendToPage: (section: string, page: number) => void
+  sendToSummary: () => void
+}
+
+function previousButtonHandler({
+  currentPage,
+  currentSection,
+  templateSections,
+  sendToPage,
+}: changePageProps) {
+  const previousPage = currentPage - 1
+
+  if (previousPage === 0) {
+    // Will check if previous page is in a other section
+    const foundSection = templateSections.find(({ index }) => index === currentSection.index - 1)
+    foundSection
+      ? sendToPage(foundSection.code, foundSection.totalPages)
+      : console.log('Problem to load previous page (not found)!')
+  } else sendToPage(currentSection.code, previousPage)
+}
+
+function nextPageButtonHandler({
+  currentPage,
+  currentSection,
+  templateSections,
+  sendToPage,
+  sendToSummary,
+}: changePageProps) {
+  const nextPage = currentPage + 1
+  // TODO: Validation
+  if (nextPage > currentSection.totalPages) {
+    // Will check if next page is in other section
+    const foundSection = templateSections.find(({ index }) => index === currentSection.index + 1)
+    foundSection ? sendToPage(foundSection.code, 1) : sendToSummary()
+  } else sendToPage(currentSection.code, nextPage)
+}
+
+export default NavigationBox
```

Comment:
I'm assuming this is just extracted from previous ApplicationPage file?

Reply:
Pretty much. Yes - Trying to declutter the `ApplicationPage` a little by doing this, since it would already take care of so much stuff, and this is actually a separated area in the app - also the reason why is named `...Box`. Let me know if you like this naming, is it clear for you?

Reply:
Yeah, seems like a good name. We should probably go through all the components and name them for more consistency, but we'll talk it through together sometime. No rush.